### PR TITLE
tests: add more tests to splunk

### PR DESCRIPTION
**Pull request by lzap**
I started adding integration `slogtest.Run` test (only available in Go 1.22+) that executes variety of contract tests against a handler covering various hidden dragons of handlers. While I was working on it, I noticed how ugly those parameters accessed by tests are and it was actually a racey code, tho only used for testing so it was harmless.

So I removed it and prepared it for a new test which I commented out in this draft so you can see what is different. I haven’t spotted it yet.

### CODE REVIEW

File: pkg/splunk/handler.go [hunk 1]
```diff
@@ -20,10 +20,18 @@ const (
...
 	jh     slog.Handler
 }
 
+// TODO might be unused
+type eventer interface {
+	event(b []byte) (int, error)
+	flush()
+	close()
+	statistics() Stats
+}
```
Comment (lzap) [inline pkg/splunk/handler.go:33]: I will remove this, found a way how to test this handler without an interface and a mock.

File: pkg/splunk/client.go [hunk 2]
```diff
@@ -93,6 +92,10 @@ func newSplunkLogger(ctx context.Context, url, token, source, hostname string) *
 		},
 	}
 
+	if maximumSize != 0 {
+		sl.maximumSize = maximumSize
+	}
+
 	rcl.RetryWaitMin = 300 * time.Millisecond
 	rcl.RetryWaitMax = 3 * time.Second
 	rcl.RetryMax = 5
```
Comment (lzap) [inline pkg/splunk/client.go:101]: I think we will ever only need to change size, maybe event size but I like your `strings.Repeat` trick so that will work. Channel lentgh should not be configurable, it does not matter too much.

File: pkg/splunk/client.go [hunk 3]
```diff
@@ -93,6 +92,10 @@ func newSplunkLogger(ctx context.Context, url, token, source, hostname string) *
 		},
 	}
 
+	if maximumSize != 0 {
+		sl.maximumSize = maximumSize
+	}
+
```
Comment (lzap) [inline pkg/splunk/client.go:98]: Since this is not a public API, I am thinking a little bit clunky behavior - maximumSize must be passed, but when 0 is passed use the default value. This works nicely with the public API which has `SplunkConfig` which defaults to zero value if you skip it.

**Comment by lzap** [comment]
I made many comments but then rebased and all of them are vanished, there you should fine the line which fixed the race detector I think. This is now ready for review.

File: pkg/splunk/handler_test.go [hunk 4]
```diff
@@ -151,3 +158,39 @@ func TestSplunkHandlerBatching(t *testing.T) {
...
+		buf := <-messages
+		err := json.Unmarshal(buf, &m)
+		if err != nil {
+			t.Fatalf("invalid json: %v\n%s", err, string(buf))
+		}
+
+		// unwrap the event
+		return m["event"].(map[string]any)["message"].(map[string]any)
+	})
+}
```
Comment (lzap) [inline pkg/splunk/handler_test.go:196]: This is a test from the standard library that performs about two dozens of various corner cases.

**Review by schuellerf — APPROVED** [review]
nice, LGTM

**Comment by lzap** [comment]
Thanks for the race found, luckily it was minor (test only) but still a good idea to keep this enabled on CICD. Merging.
